src/graphics/shader.rs

```diff
@@ -113,6 +115,18 @@ impl From<BlendMode> for Blend {
                     destination: Factor::OneMinus(BlendValue::SourceAlpha),
                 },
             },
+            BlendMode::Premultiplied => Blend {
+                color: BlendChannel {
+                    equation: Equation::Add,
+                    source: Factor::One,
```

Comment:
Still wrapping my head around it but I think these two lines are still wrong. I think the equation should be ` col_target = col_src * a_src + col_dest * a_dest`

Reply:
No, it really shouldn't be. Just imagine having a white opaque destination.
If you used this equation then this destination would never change, it would always stay white, no matter what you draw upon it.

Reply:
OK I believe you

Reply:
I think it's already right the way it is.
The new color value has to be the source color value plus whatever is left from the background/destination based upon how strong the alpha of the source is, meaning how much it covers up the background.